docs/32-building-containers.md

```diff
@@ -0,0 +1,183 @@
+
+### Custom version of R
+If you have your own R runtime, or want to use something other than the default version of R that doAzureParallel uses, you can easily point to an existing Docker image or build one yourself. This allows for the flexibility to use any R version you need without being subjected to what defaults are used by this toolkit.
+
+### Custom packages pre-built into your environment
+Installing packages is often complex and involved and takes a few tries to get right. Using docker you can make sure that your images are built correctly on your local machine without needing to try building and rebuilding doAzureParallel clusters trying to get it right. This also means that you can pull in your own custom packages and guarantee that the version of the package inside the container will never change and your runs will always produce the same results.
+
+
+### Improved cluster provisioning reliability and start up time
+One issue with installing packages is that they can take time to load and install, and are subject to potential issues with repository access and network reliability. By pre-packaging everything into your container, you can guarantee that everything is already built and available and will be loaded correctly in the doAzureParallel cluster.
+
+## Building your own container image
+Building container images may seem a bit difficult to begin with, but they are really no harder than running commands in your command line. The following sections will go through how to build a container image that will install a few R packages and their operating system dependencies.
+
+In the following example we will create an image that installs the popular web based packages jsonlite and httr. This example simply uses an image provided by the RStudio team 'r-ver' and installs a few packages into it. The benefit of using the r-ver package is that it has already done all the hard work of getting R installed, so all we need to do in add the packages we want to use and we should be good to go.
+
+NOTE: Rocker has [several great R container images](https://github.com/rocker-org/rocker/wiki) available on Docker Hub. Take a quick look through them to see if any of them suit your needs.
+
+Create a Dockerfile in a direcotry called 'demo'. Notice the Dockerfile has no extension.
+
+```sh
+mkdir demo
+touch demo/Dockerfile
+```
+
+Open up the Dockerfile with your favorite editor and paste in the following code.
+
+```Dockerfile
+# Use rocker/r-ver as the base image
+# This will inherit everyhing that was installed into base image already
+# Documented at https://hub.docker.com/r/rocker/r-ver/~/dockerfile/
+FROM rocker/r-ver
+
+# Install any dependencies required for the R packages
+RUN  apt-get update \
+  && apt-get install -y --no-install-recommends \
+  libxml2-dev
+
+
+# Install the R Packages from CRAN
+RUN Rscript -e 'install.packages(c(jsonlite, httr), dependecies = TRUE)'
```

Comment:
typo: dependecies should be dependencies

Reply:
I have removed that bit of code.